apps/chat/src/store/conversations/conversations.selectors.ts

```diff
@@ -255,6 +255,20 @@ export const selectAreSelectedConversationsExternal = createSelector(
   },
 );
 
+export const selectExternalEntities = createSelector(
+  [selectFolders, selectConversations, (state) => state],
+  (folders, conversations, state) => {
+    return {
+      externalConversations: conversations.filter((prompt) =>
+        isEntityOrParentsExternal(state, prompt, FeatureType.Chat),
+      ),
+      externalFolders: folders.filter((folder) =>
+        isEntityOrParentsExternal(state, folder, FeatureType.Chat),
+      ),
+    };
+  },
+);
```

Comment:
Why do we do all this check on UI?
Let's memorise it in selector `selectDoesAnyMyItemExists` which returns boolean instead
```suggestion
export const selectDoesAnyMyItemExist = createSelector(
  [selectFolders, selectConversations],
  (folders, conversations) => {
    const conversationRootId = getConversationRootId();
    return (
      conversations.some((conv) => conv.id.startsWith(conversationRootId)) ||
      folders.some((folder) => folder.id.startsWith(conversationRootId))
    );
  },
);
```